lib/fs/metadata.go

```diff
@@ -125,7 +128,7 @@ func (m Metadata) Marshal(out io.Writer) {
 	enc.EncodeString(magic_UTF8, "t") // type
 	enc.EncodeInt(int64(m.Typeflag))
 	enc.EncodeString(magic_UTF8, "m") // mode -- note this is *not* `os.FileMode`, it's just the perm bits
-	enc.EncodeInt(m.Mode)
+	enc.EncodeInt(m.Mode & 07777)
```

Comment:
Yeah, sounds like a good ticket to file. Within the next few weeks we should trend towards tracking these issues more formally; we might not be there yet, but closer.